src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,114 @@
+package duke;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructs an empty list of tasks.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a list of tasks using imported raw string from the file system.
+     *
+     * @param s the raw string loaded from the file system.
+     * @throws DukeException if the raw string cannot be formatted into tasks.
+     */
+    public TaskList(String s) throws DukeException {
+        String[] lines = s.split("\n");
+        tasks = new ArrayList<>();
+        for (String line : lines) {
+            addTask(Task.fromImportFormat(line));
+        }
+    }
+
+    /**
+     * Gets the number of tasks in the list.
+     *
+     * @return the number of tasks in the list.
+     */
+    public int getSize() {
```

Comment:
Nice encapsulation of the TaskList with methods getSize(), getTask(), etc.

Reply:
Good Job!